Neos.Flow/Classes/Http/Cookie.php

```diff
@@ -122,7 +122,7 @@ class Cookie
      * @api
      * @throws \InvalidArgumentException
      */
-    public function __construct($name, $value = null, $expires = 0, $maximumAge = null, $domain = null, $path = '/', $secure = false, $httpOnly = true, $sameSite = null)
+    public function __construct($name, $value = null, $expires = 0, $maximumAge = null, $domain = null, $path = '/', $secure = false, $httpOnly = true, $sameSite = 'lax')
```

Comment:
Changing the method signature is a bit more breaking than it has to be and a "magic default value" it is considered bad practice anyways (see https://github.com/kalessil/phpinspectionsea/blob/master/docs/code-style.md#non-null-parameters-default-value)

Reply:
Instead of the change here, you could adjust line 158 to
```php
$this->sameSite = $sameSite ?? self::SAMESITE_LAX
```

Reply:
I'm not soooo sure about the "magic default value" argument.

At least I find
```php
    /* original code: logic is mixed in the signature, not possible to use nullable types */
    function calculateRadial($radius, $pi = M_PI, $multiplier = 1) {
        return $radius * $pi * $multiplier;
    }
```
easier to read an grasp on a single look than
```php
    /* code refactoring step 1 (triggered by the inspection) */
    function calculateRadial($radius, $pi = null, $multiplier = null) {
        return $radius * ($pi ?? M_PI) * ($multiplier ?? 1);
    }
 ```

The former IMO better separates the algorithm from the (default) values. Default values are part of the API, not of the implementation.

Reply:
But in line 158 it makes no sense because you will end in the exception earlier ;)
I changed the exception code back and set `$sameSite` before the exception to `lax` if the `$sameSite` is `null`.

Reply:
I also find the reasoning & example a bit weird but I agree to the practice: We had quite a lot of regressions and needless breaking changes due to updates in method signatures that actually just reflected the _behavior_.

Anyways, feel free to ignore the remark in this case. We already use tons of optional arguments here and it's quite unlikely that someone extended the `Cookie` class